glide-core/src/client/value_conversion.rs

```diff
@@ -180,6 +181,35 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        ExpectedReturnType::ArrayOfDoubles => match value {
+            Value::Nil => Ok(value),
+            Value::Array(array) => {
+                let array_of_doubles = array
+                    .iter()
+                    .map(|v| match v {
+                        Value::Nil | Value::Double(_) => Ok(v.clone()),
+                        Value::BulkString(_) => Ok(convert_to_expected_type(
+                            v.clone(),
+                            Some(ExpectedReturnType::Double),
```

Comment:
you can recursivly call `convert_to_expected_type` for any type of the value there - remove `match` statement. It does all checks for you

Reply:
use `ExpectedReturnType::DoubleOrNull`